src/receive_imf.rs

```diff
@@ -1021,7 +1030,9 @@ async fn add_parts(
         state = if seen
             || fetching_existing_messages
             || is_mdn
-            || is_reaction
+            // Currently only reactions are hidden, so this check is just in case we have other
+            // hidden messages in the future to make `chat::marknoticed_chat()` no-op for them.
+            || (hidden && !is_reaction)
```

Comment:
Note(non-blocking): I myself don't think there is much point in writing it this way just in case we add future hidden messages which are not supposed to be hidden, because a) when we make predictions about the future (without any specific plan or reason to assume them), they usually turn out wrong, and b) if we do add more hidden messages then they might as well also be InFresh - we have no way of knowing about this today.

(BTW, technically the comment is wrong because location updates are also hidden)

(Also, BTW, not sure if that's obvious, so I'll explain it: Sometimes I'm writing "(blocking)" for comments that I think need to be addressed or "(non-blocking)" for comments where I think it would be nicer to change something, but if you don't think so, I would be fine with merging as-is. I'm not consistently doing that, but whenever I think it makes sense to explicitly state it.)

Reply:
I can remove this line, but the comment should be left, otherwise it's not clear why we make hidden messages `InFresh` (because they are reactions). But i prefer to add such defensive code because currently we don't know if we want `marknoticed_chat()` to deal with other kinds of hidden messages if they appear in the future, but i'd say that by default we don't want.

As for location updates, they are not hidden messages technically, they just go the trash chat.

Reply:
Removed this unnecessary check, just left a comment.